pkg/gardenlet/operation/botanist/kubeapiserverexposure.go

```diff
@@ -76,20 +76,31 @@ func (b *Botanist) DefaultKubeAPIServerSNI() component.DeployWaiter {
 		b.Shoot.ControlPlaneNamespace,
 		b.SecretsManager,
 		func() *kubeapiserverexposure.SNIValues {
-			var wildcardHosts []string
+			var wildcardConfiguration *kubeapiserverexposure.WildcardConfiguration
 
 			if b.ControlPlaneWildcardCert != nil {
-				wildcardHosts = []string{b.ComputeKubeAPIServerHost()}
+				wildcardConfiguration = &kubeapiserverexposure.WildcardConfiguration{
+					Hosts:     []string{b.ComputeKubeAPIServerHost()},
+					TLSSecret: *b.ControlPlaneWildcardCert,
+				}
+
+				// Wildcard endpoint must use the default istio ingress gateway if the shoot uses zonal istio ingress gateway.
+				// Otherwise, the wildcard endpoint can share the same istio ingress gateway as the kube-apiserver endpoint.
+				if b.DefaultIstioNamespace() != b.IstioNamespace() {
+					wildcardConfiguration.IstioIngressGateway = &kubeapiserverexposure.IstioIngressGateway{
+						Namespace: b.DefaultIstioNamespace(),
+						Labels:    b.DefaultIstioLabels(),
+					}
+				}
```

Comment:
This logic combined with the implicit assumptions in `sni.go` that the namespaces are different is somewhat hard to comprehend. Let's hope we can get rid of this special casing soon :-) .

Reply:
There are so many different cases, that's why I finally ended up in comparing the istio namespaces.

Initially, I checked if the shoot is non HA-multi-zone, but this is only correct on a multi-zone HA seed. On a non-HA seed there is only one Istio deployment.